src/js/components/hocs.js

```diff
@@ -7,15 +8,42 @@ import { deepMerge } from '../utils';
 
 export const withFocus = (WrappedComponent) => {
   class FocusableComponent extends Component {
-    state = {
-      mouseActive: false,
-      focus: false,
+    constructor() {
+      super();
+      this.handleActiveMouse = this.handleActiveMouse.bind(this);
+      this.setFocus = this.setFocus.bind(this);
+      this.state = {
+        mouseActive: false,
+        focus: false,
+      };
     }
-    setMouseActive() {
-      this.setState({ mouseActive: true });
+    componentDidMount() {
+      window.addEventListener('mousedown', this.handleActiveMouse);
+      const wrapperNode = findDOMNode(this.wrapperRef);
+      if (wrapperNode && wrapperNode.addEventListener) {
+        wrapperNode.addEventListener('focus', this.setFocus);
```

Comment:
Why do we listen for focus on the wrapper node? What if an element inside this node gets focus, for instance, the real `<input />` in a `TextInput`?